app/src/main/java/com/yacgroup/yacguide/TableActivity.kt

```diff
@@ -114,43 +115,38 @@ abstract class TableActivity : BaseNavigationActivity() {
 
     private fun _showComments(comments: List<Comment>) {
         if (comments.isNotEmpty()) {
-            val dialog = _prepareCommentDialog()
-            val layout = dialog.findViewById<LinearLayout>(R.id.commentLayout)
+            val dialog = DialogWidgetBuilder(this, R.string.comments).apply {
+                setPositiveButton { dialog, _ -> dialog.dismiss() }
+            }.create()
+            val commentDialogBinding = CommentDialogBinding.inflate(dialog.layoutInflater)
+            dialog.setView(commentDialogBinding.root)
+
             val dividerResource = TypedValue()
             theme.resolveAttribute(android.R.attr.listDivider, dividerResource, true)
             comments.forEach { comment ->
-                dialog.layoutInflater.inflate(R.layout.comment, layout, false).let { view ->
-                    val commentLayout = view.findViewById<LinearLayout>(R.id.singleCommentLayout)
-                    commentLayout.findViewById<View>(R.id.commentDivider).setBackgroundResource(dividerResource.resourceId)
+                CommentBinding.inflate(dialog.layoutInflater, commentDialogBinding.commentLayout, false).let { view ->
+                    view.commentDivider.setBackgroundResource(dividerResource.resourceId)
                     comment.properties.forEach { prop ->
                         _property2String(prop.qualityMap, prop.qualityId)?.let {
-                            val propertyLayout = commentLayout.findViewById<ConstraintLayout>(prop.layoutResource)
-                            propertyLayout.findViewById<TextView>(R.id.propertyNameTextView).setText(prop.nameResource)
-                            propertyLayout.findViewById<TextView>(R.id.propertyValueTextView).text = it
-                            propertyLayout.visibility = View.VISIBLE
+                            view.root.findViewById<ConstraintLayout>(prop.layoutResource).apply {
```

Comment:
Shouldn't the dialog hold its binding itself and you apply changes to that binding? Doing so, you could get rid of this `findViewById` I guess

Reply:
The problem is that the current implementation accesses dynamically different layouts and the code iterates over all properties. Without unrolling the loop, I don't see a way to get rid of this.